packages/offix-datastore/src/DataStore.ts

```diff
@@ -0,0 +1,39 @@
+import { buildSchema } from "graphql";
+import { readFileSync } from "fs";
+import { createDefaultStorage, Storage } from "./storage";
+import { extractModelsFromSchema, Model, PersistedModel } from "./models";
+import { createPredicate } from "./predicates";
+
+let storage: Storage;
+
+export function configure(schemaLocation: string, schemaVersion: number = 1) {
+    const schemaText = readFileSync(schemaLocation, "utf8");
+    const schema = buildSchema(schemaText);
+    const models = extractModelsFromSchema(schema);
+    storage = createDefaultStorage(models, schemaVersion);
+}
+
+export function save(model: Model): Promise<PersistedModel> {
+    return storage.save(model);
+}
+
+export function query(model: Model, predicateFunction?: Function) {
```

Comment:
So in this approach we tradig off ability to issue GraphQL Queries towards our own predicate function? Would you see ability to retrofit GraphQL queries or this is something completely different.

Reply:
AWS uses something similar. Currently, it is not possible to run graphql queries against the DataStore. Although, this is something we can add.